x-pack/qa/oidc-op-tests/build.gradle

```diff
@@ -0,0 +1,83 @@
+Project idpFixtureProject = xpackProject("test:idp-fixture")
+
+apply plugin: 'elasticsearch.standalone-rest-test'
+apply plugin: 'elasticsearch.rest-test'
+apply plugin: 'elasticsearch.test.fixtures'
+
+dependencies {
+    // "org.elasticsearch.plugin:x-pack-core:${version}" doesn't work with idea because the testArtifacts are also here
+    testCompile project(path: xpackModule('core'), configuration: 'default')
+    testCompile project(path: xpackModule('core'), configuration: 'testArtifacts')
+    testCompile project(path: xpackModule('security'), configuration: 'testArtifacts')
+}
+testFixtures.useFixture ":x-pack:test:idp-fixture"
+
+String ephemeralPort;
+task setupPorts {
+    dependsOn idpFixtureProject.postProcessFixture
+    doLast {
+        ephemeralPort = idpFixtureProject.postProcessFixture.ext."test.fixtures.oidc-provider.tcp.8080"
+    }
+// Don't attempt to get ephemeral ports when Docker is not available
+setupPorts.onlyIf { idpFixtureProject.postProcessFixture.enabled }
```

Comment:
shouldn't this line be outside of the task itself?

Reply:
:man_facepalming: copy-paste woes.

Reply:
I think it's in the right place, but the indentation is off and `setupPorts` should not be repeated as it's still in the task configuration. It's good to have a single block that configures the task fully.

Reply:
@atorok you mean something like 

```
task setupPorts {
    onlyIf { idpFixtureProject.postProcessFixture.enabled }
    dependsOn idpFixtureProject.postProcessFixture
    doLast {
        ephemeralPort = idpFixtureProject.postProcessFixture.ext."test.fixtures.oidc-provider.tcp.8080"
    }
}
```

Reply:
Ok, now that I re-read your comment it seemed much more clear to me than the fist time around, I'll change that